src/DB/Data/ConnectionFactory.php

```diff
@@ -35,27 +48,63 @@ public function __construct(DbConfig $dbConfig)
      *
      * @param string $connectionType
      * @return ConnectionInterface
-     * @throws \RuntimeException
+     * @throws RuntimeException
+     * @throws ConfigException
      */
     public function create(string $connectionType): ConnectionInterface
     {
         switch ($connectionType) {
             case self::CONNECTION_MAIN:
-                $connectionData = $this->dbConfig->get()['connection']['default'] ?? [];
-                $connection = new Connection($connectionData);
-                break;
+                return $this->getConnectionData(DbConfig::CONNECTION_DEFAULT);
             case self::CONNECTION_SLAVE:
-                $connectionData = $this->dbConfig->get()['slave_connection']['default']
-                    ?? $this->dbConfig->get()['connection']['default']
-                    ?? [];
-                $connection = new Connection($connectionData);
-                break;
+                return $this->getConnectionData(DbConfig::CONNECTION_DEFAULT, false);
+            case self::CONNECTION_QUOTE_MAIN:
+                return $this->getConnectionData(DbConfig::CONNECTION_CHECKOUT);
+            case self::CONNECTION_QUOTE_SLAVE:
+                return $this->getConnectionData(DbConfig::CONNECTION_CHECKOUT, false);
+            case self::CONNECTION_SALES_MAIN:
+                return $this->getConnectionData(DbConfig::CONNECTION_SALES);
+            case self::CONNECTION_SALES_SLAVE:
+                return $this->getConnectionData(DbConfig::CONNECTION_SALES, false);
             default:
-                throw new \RuntimeException(
+                throw new RuntimeException(
                     sprintf('Connection with type %s doesn\'t exist', $connectionType)
                 );
         }
+    }
+
+    /**
+     * Returns connection data by name
+     *
+     * @param string $name
+     * @param bool $isMain
+     *
+     * @return ConnectionInterface
+     * @throws ConfigException
+     */
+    private function getConnectionData(string $name, bool $isMain = true): ConnectionInterface
```

Comment:
Instead of passing the second boolean argument I would recommend to create two methods:
`getMainConnectionData()` and `getSlaveConnectionData`

Reply:
If this was a public method I would agree with you, but do you think it's worth splitting it on a private method?

Reply:
As for me, this is not important because it is a private implementation